### PR TITLE
Tutorial on how to convert a protobuf message and filter certain fields to a resulting map.

### PR DESCRIPTION
**Pull request by marcoshuck**
This PR addresses the discussion raised in #41 

I ended up copying the `CamelCase` code from the function to my personal open source project to avoid code duplications due to the limitation of the github protobuf project being deprecated, we can remove the dependency if it doesn't make sense to keep it in here.

### CODE REVIEW

**Comment by marcoshuck** [comment]
Seems CI pipelines are failing, looking into it.

File: README.md
Comment (mennanov) [inline README.md:1]: Hm... i don't think that bringing a new dependency is justified in this case.
The purpose of this callback function is to provide a mapping from the proto field names to the golang struct field names. They may not even match (e.g. "a" can be mapped to "b").

I would just add more details to the `README.md` file: extend the existing examples with a real callback function to make it clear how it is supposed to be used.

I hope this makes sense.
Thanks!

**Review by mennanov — CHANGES_REQUESTED** [review]

**Comment by mennanov** [comment]
> Seems CI pipelines are failing, looking into it.

Yeah, i think i need to upgrade the dependencies and go version. I'll do that in a separate PR.

File: README.md
Comment (marcoshuck) [reply README.md:1]: It does, I'll remove the Go code and improve the README

**Comment by mennanov** [comment]
> > Seems CI pipelines are failing, looking into it.
> 
> Yeah, i think i need to upgrade the dependencies and the go version. I'll do that in a separate PR.

I ended up not upgrading the go version in `go.mod` since it may impact the existing users of the library as according to the docs (https://go.dev/doc/modules/gomod-ref#go):

> Go toolchains refuse to use modules declaring newer Go versions.

I updated Github Actions tests for go 1.21 though.

File: README.md
Comment (marcoshuck) [reply README.md:1]: I've removed the dependency 😄

File: go.sum
Comment (mennanov) [inline go.sum:1]: lease revert the changes in this file.

**Review by mennanov — COMMENTED** [review]
Looks good overall, but please revert the `go.sum` changes

File: go.sum
Comment (marcoshuck) [reply go.sum:1]: Interesting, seems I forgot to run `go mod tidy`. Done!

**Review by mennanov — APPROVED** [review]
Looks good!

**Comment by mennanov** [comment]
Thanks for contributing!

**Pull request merged**